sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_pyamqp_transport.py

```diff
@@ -0,0 +1,994 @@
+        """
+        Receiver enhanced_message_received callback.
+        """
+        # pylint: disable=protected-access
+        receiver._handler._last_activity_timestamp = time.time()
+        if receiver._receive_context.is_set():
+            receiver._handler._received_messages.put((frame, message))
+        else:
+            receiver._handler.settle_messages(frame[1], 'released')
+
+    @staticmethod
+    def build_received_message(
+        receiver: "ServiceBusReceiver",
+        message_type: "ServiceBusReceivedMessage",
+        received: "Message"
+    ):
+        # pylint: disable=protected-access
+        message = message_type(
+            message=received[1], receive_mode=receiver._receive_mode, receiver=receiver, frame=received[0]
+        )
+        receiver._last_received_sequenced_number = message.sequence_number
+        return message
+
+    @staticmethod
+    def get_receive_timeout(handler):
+        """
+        Gets the timeout on the ReceiveClient.
+        :param ReceiveClient handler: Handler to set timeout on.
+        :rtype: int
+        """
+        return handler._idle_timeout
+
+    @staticmethod
+    def set_receive_timeout(handler, max_wait_time):
+        """
+        Sets the timeout on the ReceiveClient and returns original timeout.
+        :param ReceiveClient handler: Handler to set timeout on.
+        :param int max_wait_time: Max wait time.
+        :rtype: None
+        """
+        # _timeout to _idle_timeout
+        handler._idle_timeout = max_wait_time
+
+    @staticmethod
+    def get_current_time(handler):  # pylint: disable=unused-argument
+        """
+        Gets the current time.
+        :param ReceiveClient handler: Handler with counter to get time. Unused for pyamqp.
+        :rtype: int
+        """
+        return time.time()
+
+    @staticmethod
+    def reset_link_credit(handler, link_credit):
+        """
+        Resets the link credit on the link.
+        :param ReceiveClient handler: Client with link to reset link credit.
+        :param int link_credit: Link credit needed.
+        :rtype: None
+        """
+        handler._link.flow(link_credit=link_credit) # pylint: disable=protected-access
+
+    @staticmethod
+    def settle_message_via_receiver_link(
+        handler: ReceiveClient,
+        message: "ServiceBusReceivedMessage",
+        settle_operation: str,
+        dead_letter_reason: Optional[str] = None,
+        dead_letter_error_description: Optional[str] = None,
+    ) -> None:
+        if settle_operation == MESSAGE_COMPLETE:
+            return handler.settle_messages(message.delivery_id, 'accepted')
+        if settle_operation == MESSAGE_ABANDON:
+            return handler.settle_messages(
+                message.delivery_id,
+                'modified',
+                delivery_failed=True,
+                undeliverable_here=False
+            )
+        if settle_operation == MESSAGE_DEAD_LETTER:
+            return handler.settle_messages(
```

Comment:
handler.settle_messages doesn't return anything, so this will return None anyway. I think these are here to it doesn't raise the ValueError at the bottom of the method.